drivers/ethernet/eth_nxp_enet.c

```diff
@@ -632,6 +625,22 @@ static int nxp_enet_phy_init(const struct device *dev)
  ****************************
  */
 
+void nxp_enet_driver_cb(const struct device *dev,
+				enum nxp_enet_driver dev_type,
+				enum nxp_enet_callback_reason event,
+				void *data)
+{
+	if (dev_type == NXP_ENET_MDIO) {
+		nxp_enet_mdio_callback(dev, event, data);
+	}
+
+#ifdef CONFIG_PTP_CLOCK_NXP_ENET
+	if (dev_type == NXP_ENET_PTP_CLOCK) {
+		nxp_enet_ptp_clock_callback(dev, event, data);
+	}
+#endif
```

Comment:
Only one type will probably be handled?

```c
	switch (dev_type) {
	case NXP_ENET_MDIO:
		nxp_enet_mdio_callback(dev, event, data);
		break;
#ifdef CONFIG_PTP_CLOCK_NXP_ENET
	case NXP_ENET_PTP_CLOCK:
		nxp_enet_ptp_clock_callback(dev, event, data);
		break;
#endif
	}
```

Reply:
I believe the compiler will optimize what I have here, I don't like switch case in source code except for state machines, personally